docs/tutorials/podman_pasta_ipv6_with_snid_ipv4.md

```diff
@@ -0,0 +1,761 @@
+
+References:
+- https://storychief.io/blog/301-302-307-308-redirect
+- https://www.infidigit.com/blog/308-permanent-redirect/
+
+When a 308 redirect code is specified, the client must repeat the exact same request (POST or GET) on the target location. For 301 redirect, the client may not necessarily follow the exact same request. 
+
+Since the HTTPS must always be Enforced (`Permanent Redirect`) and the Client should perform the exact same Request over HTTPS that was performed over HTTP, the Code `308` seems more Appropriate.
+
+## HTTP -> HTTPS for IPv4 Redirects
+The `snid` Service does NOT handle HTTP (non-HTTPS) Requests and does NOT bind to the IPv4 Address on Port 80.
+
+This Problem (IPv4 HTTP -> HTTPS Redirects) can easily be solved by using a one-off (for the entire Podman Host) Caddy Container.
+
+(Option 1) `compose.yml` with "Normal" Ports Mappings:
+```
+services:
+  redirect-http-ipv4-caddy:
+    image: caddy:latest
+    pull_policy: "missing"
+    container_name: redirect-http-ipv4-caddy
+    restart: "unless-stopped"
+    security_opt:
+      - no-new-privileges:true
+      - label=type:container_runtime_t
+    ports:
+      - target: 80
+        host_ip: 172.16.1.10
+        published: 80
+        protocol: tcp
+    network_mode: "pasta:--ipv4-only"
+    volumes:
+      - ./Caddyfile:/etc/caddy/Caddyfile:ro,z
+      - ~/containers/data/redirect-http-ipv4-caddy:/data:rw,z
+      - ~/containers/log/redirect-http-ipv4-caddy:/var/log:rw,z
+      - ~/containers/config/redirect-http-ipv4-caddy:/config:rw,z
+      - ~/containers/certificates/letsencrypt:/certificates:ro,z
+    environment:
+      - CADDY_DOCKER_CADDYFILE_PATH=/etc/caddy/Caddyfile
+
+```
+
+(Option 2) `compose.yml` with "Pasta" One-Liner:
+```
+services:
+  redirect-http-ipv4-caddy:
+    image: caddy:latest
+    pull_policy: "missing"
+    container_name: redirect-http-ipv4-caddy
+    restart: "unless-stopped"
+    security_opt:
+      - no-new-privileges:true
+      - label=type:container_runtime_t
+    network_mode: "pasta:--ipv4-only,-t,172.16.1.10/80"
+    volumes:
+      - ./Caddyfile:/etc/caddy/Caddyfile:ro,z
+      - ~/containers/data/redirect-http-ipv4-caddy:/data:rw,z
+      - ~/containers/log/redirect-http-ipv4-caddy:/var/log:rw,z
+      - ~/containers/config/redirect-http-ipv4-caddy:/config:rw,z
+      - ~/containers/certificates/letsencrypt:/certificates:ro,z
+    environment:
+      - CADDY_DOCKER_CADDYFILE_PATH=/etc/caddy/Caddyfile
+```
+
+
+`Caddyfile`:
+```
+# Example and Guide
+# https://caddyserver.com/docs/caddyfile/options
+
+# General Options
+{
+    # Debug Mode
+    debug
+}
+
+# Redirect TCP IPV4 HTTP Requests to HTTPS
+http:// {
+  bind tcp4/172.16.1.10
+  redir https://{host}{uri} 308
+
+# Logging
+  log {
+	output file /var/log/access.json {
+		roll_size 100MiB
+		roll_keep 5000
+		roll_keep_for 720h
+		roll_uncompressed
+	}
+        format json
+  }
+}
+```
+
+## HTTP -> HTTPs for IPv6 Redirects
```

Comment:
i'm not so sure this section is needed, or in such detail.

Reply:
Reduced to:

## HTTP -> HTTPs for IPv6 Redirects
"Native" IPv6 Redirects are automatically Handled by Caddy Proxy listening on the Configured IPv6 Address.
See [Caddy Documentation](https://caddyserver.com/docs/automatic-https).